docs/DeveloperGuide.md

```diff
@@ -168,6 +170,44 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Keyboard Navigation Feature
+
+#### Overview
+
+The keyboard navigation feature allows users to navigate between different UI components using predefined keyboard shortcuts. This feature enhances fast-typing users' experience by allowing them to navigate the UI without having to use the mouse.
+
+#### Implementation
+
+The implementation of this feature is centered in the `MainWindow` class, which acts as the primary UI container for the application. Within `MainWindow`, two key methods are used: `setKeyNavigation` and `setKeyNavigations`. The `setKeyNavigations` utilizes `setKeyNavigation` to assign specific keyboard events to different UI parts.
+
+Here's the implementation of those method:
+
+```java
+private void setKeyNavigation(UiPartFocusable<?> uiPartFocusable, KeyCombination keyCombination) {
+    getRoot().addEventFilter(KeyEvent.KEY_PRESSED, event -> {
+        if (keyCombination.match(event)) {
+            if (focusedUiPart != null) {
+                focusedUiPart.unFocus();
+            }
+            uiPartFocusable.focus();
+            focusedUiPart = uiPartFocusable;
+            event.consume();
+        }
+    });
+}
+```
+
+```java
+void setKeyNavigations() {
+    setKeyNavigation(personListPanel, KeyCombination.keyCombination("Alt+LEFT"));
+    setKeyNavigation(transactionListPanel, KeyCombination.keyCombination("Alt+RIGHT"));
+    setKeyNavigation(resultDisplay, KeyCombination.keyCombination("Alt+UP"));
+    setKeyNavigation(commandBox, KeyCombination.keyCombination("Alt+DOWN"));
```

Comment:
I think let's leave out the code in the UG, I remember somewhere in the textbook mentioning that it is not advised to do so. Perhaps we can replace this part with just 1 example

Reply:
@jeffrey-jian I just updated the section. What do you think?